core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java

```diff
@@ -79,18 +84,23 @@ public FieldSortBuilder randomFieldSortBuilder() {
         if (randomBoolean()) {
             builder.sortMode(randomFrom(SortMode.values()));
         }
-
```

Comment:
Shouldn't we test only three cases: no_sort, new_sort, old_sort ? Mixing the old and the new format should not be allowed.

Reply:
I changed this to either test one or the other. I can also add checks on the setters that prohibit setting both version at the same time, but I didn't want to change too much of the existing behaviour of the class under test in this PR, wdyt?

Reply:
I added an additional commit to this PR that changes the current behaviour to throw an error if both formats are used at the same time. If you think this is good we can change the title of the PR to not only be about testing.

Reply:
++ to do that